documentation/modules/configuring/con-config-mirrormaker2-producers-consumers.adoc

```diff
@@ -0,0 +1,109 @@
+// Module included in the following assemblies:
+//
+// assembly-config-mirrormaker2.adoc
+
+[id='con-mirrormaker-producers-consumers-{context}']
+= Connector producer and consumer configuration
+
+[role="_abstract"]
+MirrorMaker 2.0 connectors uses internal producers and consumers.
+If needed, you can configure these producers and consumers to override the default settings. 
+
+For example, you can increase the `batch.size` for the source producer that sends topics to the target Kafka cluster to better accommodate large volumes of messages.
+
+IMPORTANT: Producer and consumer configuration options depend on the MirrorMaker 2.0 implementation, and may be subject to change.  
+
+The following tables describe the producers and consumers for each of the connectors and where you can add configuration. 
+
+.Source connector producers and consumers
+[cols="1,1a,3m",options="header"]
+|===
+
+|Type
+|Description
+|Configuration
+
+|Producer
+|Sends topic messages to the target Kafka cluster. Consider tuning the configuration of this producer when it is handling large volumes of data. 
+|mirrors.sourceConnector.config: producer.override.*
+
+|Producer
+|Writes to the `offset-syncs` topic, which maps the source and target offsets for replicated topic partitions. 
+|mirrors.sourceConnector.config: producer.*
+
+|Consumer
+|Retrieves topic messages from the source Kafka cluster.
+|mirrors.sourceConnector.config: consumer.* 
+|===
+
+.Checkpoint connector producers and consumers
+[cols="1,1a,3m",options="header"]
+|===
+
+|Type
+|Description
+|Configuration
+
+|Producer
+|Emits consumer offset checkpoints.
+|mirrors.checkpointConnector.config: producer.override.* 
+
+|Consumer
+|Loads the `offset-syncs` topic.
+|mirrors.checkpointConnector.config: consumer.*
+
+|===
+
+IMPORTANT: The consumer configuration changes depending on the location of the `offset-syncs` topic. The default location is the source Kafka cluster. You can set `offset-syncs.topic.location` to `target` to use the target Kafka cluster. If you use the target Kafka cluster, use `mirrors.checkpointConnector.config: target.consumer.*` for the consumer configuration. 
+
+.Heartbeat connector producer
+[cols="1,1a,3m",options="header"]
+|===
+
+|Type
+|Description
+|Configuration
+
+|Producer
+|Emits heartbeats.
+|mirrors.heartbeatConnector.config: producer.override.*
+
+|===
+
+.Example configuration for connector producers and consumers
+[source,yaml,subs="+quotes,attributes"]
```

Comment:
A certain number of users try to use the configurations described in the documentation as they are considering them to be the recommended values (somewhat surprisingly). It might be safer to simply use the default values for each configurations in this example. 

I think we can use the sourceConnector configurations discussed in https://github.com/strimzi/strimzi-kafka-operator/pull/7006. However, it would be safe to simply use the default values for checkpointConnector and HeartbeatConnector configuration. Here are the default values:
```
request.timeout.ms: 30000
max.poll.interval.ms: 300000
```

Reply:
Yes the default values are suitable in many cases. I think this example should be described as "this is how you do it" rather than "here are some good values you should use"

Reply:
I changed to the default values and introduced the example with the following:

> The following example shows how you configure the producers and consumers.